File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/validation/validator/UsernameValidator.java
package com.windea.demo.cloudcollect.core.validation.validator;

import com.windea.demo.cloudcollect.core.validation.annotation.Username;
import org.springframework.beans.factory.annotation.Value;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

/**
 * 用户名的校验器。
 */
public class UsernameValidator implements ConstraintValidator<Username, String> {
	@Value("${com.windea.validation.validUsername}")
	private String validUsername;


	@Override
	public boolean isValid(String value, ConstraintValidatorContext context) {
		//6~16位的字母、数字和下划线。
		return value.matches(validUsername);
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/validation/validator/UsernameValidator.kt
package com.windea.demo.cloudcollect.core.validation.validator

import com.windea.demo.cloudcollect.core.properties.*
import com.windea.demo.cloudcollect.core.validation.annotation.*
import org.springframework.beans.factory.annotation.*
import org.springframework.stereotype.*
import javax.validation.*

/**用户名的校验器。*/
@Component
class UsernameValidator : ConstraintValidator<Username, String> {
	@Autowired lateinit var validationProperties: ValidationProperties
	
	override fun isValid(value: String, context: ConstraintValidatorContext): Boolean {
		//6~16位的字母、数字和下划线。
		return value.matches(validationProperties.username.toRegex())
	}
}
